### PR TITLE
Fix clearing specific items in clearinventory

### LINKED ISSUE #2986: The EssentialsX clear command will not let me clear all items of a specific type
### Help request
**Output of /ess version**

Server version: 1.12.2-R0.1-SNAPSHOT git-Paper-1618 (MC: 1.12.2)
EssentialsX version: 2.17.1.53
LuckPerms version: 5.0.39
Vault version: 1.7.2-b107
EssentialsXProtect version: 2.17.1.53
EssentialsXChat version: 2.17.1.53
EssentialsXSpawn version: 2.17.1.53

**Problem**
Whenever I use the EssentialsX clear inventory command I cannot get it to clear ALL items of a specific type it just says: "Removed 0 of [Item] from [Username]"

**What I have tried**
I have tried using the Minecraft clear command with /Minecraft:clear but I do not want to have to use the Minecraft command as I would rather use the EssentialsX command

**Screenshots**

### PR DESCRIPTION
**Pull request by pop4959**
Fixes #2986
Closes #3050
Closes #3191

Refactors the clearinventory command code in such a way that fixes the problem of not being able to clear all of a specific item on newer server versions. Also re-adds data value support for older server versions that support it, which seemed to have been inadvertently removed in 79bc34047b86461e752491d08ee8a38a5c07d9dd.

Checked against 1.8.8, 1.12.2, 1.13.2, 1.15.2.

### CODE REVIEW

**Review by darbyjack — DISMISSED** [review]

**Review by JRoy — DISMISSED** [review]

File: Essentials/src/com/earth2me/essentials/commands/Commandclearinventory.java [hunk 1]
```diff
@@ -79,11 +80,28 @@ private void parseCommand(Server server, CommandSource sender, String commandLab
...
+            return material;
+        }
+
+        public short getData() {
+            return data;
+        }
+    }
+
+    protected void clearHandler(CommandSource sender, Player player, String[] args, int offset, boolean showExtended) {
         int type = -1;
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/commands/Commandclearinventory.java]: I'd prefer this was replaced with a private enum, as done in #3191 - this variable only exists to control flow later on, and it's clearer if it uses a named enum constant than a magic number.

**Review by mdcfe — CHANGES_REQUESTED** [review]

**Review by JRoy — APPROVED** [review]

**Review by mdcfe — APPROVED** [review]

**Comment by x3mp** [comment]
Clearing items with custom names still does not work properly?

**Comment by mdcfe** [comment]
@UnrealPixel Named items are not supported by `/clear`.

**Comment by UnrealPixel** [comment]
> @UnrealPixel Named items are not supported by `/clear`.
How to clear custom named items?

**Comment by x3mp** [comment]
@md678685 Then how do I clear items with a name?

**Comment by pop4959** [comment]
@UnrealPixel The bug that was fixed in this PR was that you could not clear *specific* items (as in, for example only dirt), only the whole inventory, whereas in the past this was possible.

You should be able to use [/minecraft:clear](https://minecraft.gamepedia.com/Commands/clear) for removing items with specific NBT tags. The Essentials command is not designed to be so advanced.

**Comment by x3mp** [comment]
@pop4959 That is why I reopened #592 which was closed because this was the "same" issue.
when I do: /minecraft:clear @p rotten_flesh 0 1 {display:{Name:&8Scraps}} it doesn't clear the rotten flesh with the name Scraps in Gray.
It just presents me with the error:
Data tag parsing failed: Expected value at:
 {display:{Name:<--[HERE]

**Pull request merged**